components/Controllers/SwapPositionsPageContent/SwapPositionsPageContent.tsx

```diff
@@ -68,82 +69,206 @@ export function SwapPositionsPageContent() {
+import { usePoolTokens } from 'hooks/usePoolTokens';
+
+import { getActivityKind } from '../Activity/Activity';
+
+interface ActivityWithRunningBalance extends ActivityType {
+  runningPaprDebt?: number;
+  runningPaprBalance?: number;
+  paprDelta?: number;
+  ethDelta?: number;
+  kind?: string;
+}
+
+const ActivityTimeline: React.FC<ActivityTimelineProps> = ({
+  account,
+  startTime,
+  endTime,
+}) => {
+  const {
+    data: activityData,
+    fetching: activityFetching,
+    fetchMore,
+    remaining,
+  } = useActivity(
+    '0x3b29c19ff2fcea0ff98d0ef5b184354d74ea74b0',
+    account,
+    undefined,
+    500,
+  );
+
+  const { implicitSwaps } = useLPActivityAndImplicitSwaps(
+    account,
+    startTime,
+    endTime,
+  );
+
+  const { token0, token1 } = usePoolTokens();
+
+  const [activities, setActivities] = useState<
+    ActivityWithRunningBalance[] | null
+  >(null);
+
+  useEffect(() => {
+    const filteredActivities = activityData.filter(
+      (a) => a.timestamp >= startTime && a.timestamp <= endTime,
+    );
+    filteredActivities.reverse();
+    let runningTotal = 0;
+    let runningDebtTotal = 0;
+    const a: ActivityWithRunningBalance[] = filteredActivities.map((d) => {
+      runningTotal += d.tokenIn
+        ? d.tokenIn.symbol == 'paprMEME'
+          ? (d.amountIn / 1e18) * -1
+          : d.amountOut / 1e18
+        : 0;
+      runningDebtTotal += d.amountBorrowed ? d.amountBorrowed / 1e18 : 0;
+      runningDebtTotal -= d.amountRepaid ? d.amountRepaid / 1e18 : 0;
+      const activity: ActivityWithRunningBalance = d;
+      activity.runningPaprBalance = runningTotal;
+      activity.runningPaprDebt = runningDebtTotal;
+      activity.kind = getActivityKind(d);
+      if (activity.liquidityDelta) {
+        const id = `${activity.id}`;
+        const implied = implicitSwaps.filter((s) => s.id == id);
+        if (implied.length > 0) {
+          const s = implied[0];
+          activity.amountIn = s.amountIn;
+          activity.amountOut = s.amountOut;
+          activity.tokenIn = s.tokenIn;
+          activity.tokenOut = s.tokenOut;
+          runningTotal += s.tokenIn
+            ? s.tokenIn.symbol == 'paprMEME'
+              ? (s.amountIn / 1e18) * -1
+              : s.amountOut / 1e18
+            : 0;
+        }
+        activity.runningPaprBalance = runningTotal;
+      }
+      return activity;
+    });
+    a.reverse();
+    setActivities(a);
+  }, [activityData, startTime, endTime, account, implicitSwaps]);
+
+  return (
+    <table className={styles.table}>
+      <thead>
+        <tr className={styles.tr}>
+          <th>Date</th>
+          <th>action type</th>
+          <th>vault</th>
+          <th>running total papr debt</th>
+          <th>papr held delta</th>
+          <th>eth held delta</th>
+          <th>running total papr bought minus sold</th>
+        </tr>
+      </thead>
+      <tbody>
+        {activities &&
+          activities.map((d) => {
```

Comment:
why variable name of d?

Reply:
datum? not sure I remember what I was doing